dali/pipeline/operators/geometric/flip.cc

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/pipeline/operators/geometric/flip.h"
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+
+DALI_SCHEMA(Flip)
+    .DocStr(R"code(Converts between various image color models)code")
+        .NumInput(1)
+        .NumOutput(1)
+        .AllowMultipleInputSets()
+        .AddOptionalArg("horizontal",
+                R"code(The color space of the input image)code",
+                1)
+        .AddOptionalArg("vertical",
+                R"code(The color space of the output image)code",
+                0);
+
+template <>
+Flip<CPUBackend>::Flip(const OpSpec& spec)
+    : Operator<CPUBackend>(spec)
+    , _horizontal(spec.GetArgument<int32>("horizontal"))
+    , _vertical(spec.GetArgument<int32>("vertical")) {}
+
+
+enum class FlipAxis {
+  HORIZONTAL,
+  VERTICAL,
+  BOTH
+};
+
+template <FlipAxis D>
+struct FlipImpl {};
+
+template <>
```

Comment:
Just wondering: Is there any reason to reimplement the flip operation instead of using the one in opencv:

https://docs.opencv.org/2.4/modules/core/doc/operations_on_arrays.html?#void%20flip(InputArray%20src,%20OutputArray%20dst,%20int%20flipCode)
?